cylc/flow/task_events_mgr.py

```diff
@@ -1533,7 +1533,18 @@ def _insert_task_job(
         if (itask.tdef.run_mode == RunMode.SIMULATION) or forced:
             job_conf = {"submit_num": itask.submit_num}
         else:
-            job_conf = itask.jobs[-1]
+            try:
+                job_conf = itask.jobs[-1]
+            except IndexError:
+                # we do not have access to the job config
```

Comment:
Can we add, to this comment, your your theory on why this might be needed? 

```suggestion
                # we do not have access to the job config
                # GitHub #6314: itask.jobs was empty on restart, perhaps because it
                # did not get written to the DB due to an emergency shutdown.
```